src/isomorphism.rs

```diff
@@ -346,13 +346,36 @@ where
     new_graph
 }
 
-/// [Graph] Return `true` if the graphs `g0` and `g1` are isomorphic.
+trait SemanticMatcher<T> {
+    fn enabled(&self) -> bool;
+    fn eq(&mut self, _: &T, _: &T) -> PyResult<bool>;
+}
+
+impl<T, F> SemanticMatcher<T> for Option<F>
+where
+    F: FnMut(&T, &T) -> PyResult<bool>,
+{
+    #[inline]
+    fn enabled(&self) -> bool {
+        self.is_some()
+    }
+    #[inline]
+    fn eq(&mut self, a: &T, b: &T) -> PyResult<bool> {
+        let res = (self.as_mut().unwrap())(a, b)?;
+        Ok(res)
+    }
+}
+
+macro_rules! impl_isomorphic {
+($is_isomorphic:ident, $try_match:ident, $op:tt) => {
+
+/// [Graph] Return `true` if the graphs `g0` and `g1` are (sub) graph isomorphic.
```

Comment:
Well, i see your point. I used a macro since it was fun, fast (required less coding) and it's not straight forward to use an `Ordering` here, since in `is_isomorphic` we would check:
```rust
a.cmp(&b) == Equal
```
but in `is_subgraph_isomorphic`:
```rust
a.cmp(&b) == Equal or Greater
```

> Is it idiomatic to use Rust macros like that?

Honestly, i don't know but i have seen similar things in petgraph.

Reply:
> Well, i see your point. I used a macro since it was fun, fast (required less coding) and it's not straight forward to use an `Ordering` here, since in `is_isomorphic` we would check:
> 
> ```rust
> a.cmp(&b) == Equal
> ```
> 
> but in `is_subgraph_isomorphic`:
> 
> ```rust
> a.cmp(&b) == Equal or Greater
> ```
> 
> > Is it idiomatic to use Rust macros like that?
> 
> Honestly, i don't know but i have seen similar things in petgraph.

You can overcome the `Equal or Greater` issue using [Ordering.then](https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then):

```rust
a.cmp(&b).then(expected_ordering) == expected_ordering
```
If `a.cmp(&b)` happens to be `Orderign::Equal`, it will be converted to `expected_ordering`. Does this help making it easier to write the code without the macro?

Reply:
Yes it does. I will update it